### PR TITLE
Add Go 3LO and API key auth support

### PR DESCRIPTION
**Pull request by saicheems**

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/discoveries/go/go_adexchangebuyer.v1.4.json.baseline [hunk 1]
```diff
@@ -6,23 +6,22 @@ package main
...
 )
 
 func main() {
   ctx := context.Background()
-  hc, err := google.DefaultClient(ctx, adexchangebuyer.AdexchangeBuyerScope)
+  // TODO: Implement this function to get authentication credentials.
+  // See https://foo.com/bar
+  // Authorize using one of the following scopes in order to use this method:
+  //   https://www.googleapis.com/auth/adexchange.buyer
+  hc, err := getClient(ctx)
```
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/testdata/discoveries/go/go_adexchangebuyer.v1.4.json.baseline]: define a stub here, as in Python
```
hc, err := func(_ *context.foo) (*bar, error) { return nil, nil}(ctx)
```

Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/go/go_adexchangebuyer.v1.4.json.baseline]: Already implemented a function, PTAL at `getClient`

**Comment by vchudnov-g** [comment]
LGTM

**Comment by saicheems** [comment]
FYI, I added a small rider to this pull that crudely fixes the semicolon comment bug where:

`... // eg. "us-central-1";`

would show up.

**Pull request merged**